test/e2e/common/pods.go

```diff
@@ -192,8 +195,35 @@ var _ = framework.KubeDescribe("Pods", func() {
 			LabelSelector:   selector.String(),
 			ResourceVersion: pods.ListMeta.ResourceVersion,
 		}
-		w, err := podClient.Watch(options)
-		Expect(err).NotTo(HaveOccurred(), "failed to set up watch")
+
+		listCompleted := make(chan bool)
+		lw := &cache.ListWatch{
+			ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+				options.LabelSelector = selector.String()
+				podList, err := podClient.List(options)
+				if err == nil {
+					timeout := time.After(30 * time.Second)
+					for {
+						select {
+						case listCompleted <- true:
+							framework.Logf("observed the pod list")
+							return podList, err
+						case <-timeout:
+							return podList, err
+						default:
+							time.Sleep(5 * time.Second)
```

Comment:
there's no need to sleep here, or to have a timeout. selecting between `listCompleted <- true` and a no-op `default` case is sufficient

Reply:
@liggitt `listCompleted` channel failing to receive the Boolean value, because of channel overflow to avoid it, added sleep and timeout.

Reply:
Create a buffered channel with an initial capacity of 1. No need to sleep in the default case.

Reply:
@liggitt Thank you, its working fine.